libtransmission/rpc-server.cc

```diff
@@ -544,6 +558,110 @@ static auto constexpr ServerStartRetryDelayIncrement = int{ 5 };
 static auto constexpr ServerStartRetryDelayStep = int{ 3 };
 static auto constexpr ServerStartRetryMaxDelay = int{ 60 };
 
+static char const* tr_rpc_address_to_string(tr_rpc_address const* addr, char* buf, size_t buflen)
+{
+    TR_ASSERT(tr_rpc_address_is_valid(addr));
+
+    switch (addr->type)
+    {
+    case TR_RPC_AF_INET:
+        return evutil_inet_ntop(AF_INET, &addr->addr, buf, buflen);
+
+    case TR_RPC_AF_INET6:
+        return evutil_inet_ntop(AF_INET6, &addr->addr, buf, buflen);
+
+    case TR_RPC_AF_UNIX:
+        tr_strlcpy(buf, addr->addr.unixSocketPath, buflen);
+        return buf;
+
+    default:
+        return nullptr;
+    }
+}
+
+static void tr_rpc_address_with_port_to_string(char* buf, size_t buflen, tr_rpc_server const* server)
+{
+    char addr_buf[TrUnixAddrStrLen];
+    tr_rpc_address const* addr = &server->bindAddress;
+
+    tr_rpc_address_to_string(addr, addr_buf, sizeof(addr_buf));
+
+    if (addr->type == TR_RPC_AF_UNIX)
+    {
+        tr_strlcpy(buf, addr_buf, buflen);
+    }
+    else
+    {
+        tr_snprintf(buf, buflen, "%s:%d", addr_buf, tr_rpcGetPort(server));
+    }
+}
+
+static bool tr_rpc_address_from_string(tr_rpc_address* dst, std::string_view src)
+{
+    if (src.compare(0, strlen(TrUnixSocketPrefix), TrUnixSocketPrefix) == 0)
+    {
+        if (src.length() >= TrUnixAddrStrLen)
+        {
+            tr_logAddNamedError(
+                MyName,
+                _("Unix socket length too long, must be less than %zu characters (including \"%s\" prefix)"),
+                TrUnixAddrStrLen - 1,
+                TrUnixSocketPrefix);
+            return false;
+        }
+
+        dst->type = TR_RPC_AF_UNIX;
+        tr_strlcpy(dst->addr.unixSocketPath, std::string{ src }.c_str(), TrUnixAddrStrLen);
+        return true;
+    }
+
+    if (evutil_inet_pton(AF_INET, std::string{ src }.c_str(), &dst->addr) == 1)
+    {
+        dst->type = TR_RPC_AF_INET;
+        return true;
+    }
+
+    if (evutil_inet_pton(AF_INET6, std::string{ src }.c_str(), &dst->addr) == 1)
+    {
+        dst->type = TR_RPC_AF_INET6;
+        return true;
+    }
+
+    return false;
+}
+
+static bool bindUnixSocket(struct event_base* base, struct evhttp* httpd, char const* path, int socket_mode)
+{
+#ifdef _WIN32
+    tr_logAddNamedError(
+        MyName,
+        _("Unix sockets are not supported on Windows, please change \"%s\" in your configuration file."),
```

Comment:
_(nit)_ run-on sentence

```suggestion
        _("Unix sockets are not supported on Windows. Please change \"%s\" in your configuration file."),
```